src/laplace.jl

```diff
@@ -16,7 +16,14 @@ copy(D::Laplacian) = Laplacian(copy(D.axis), D.k)
      # Spherical harmonics are the eigenfunctions of the Laplace operator on the unit sphere
      P * Diagonal(mortar(Fill.((-(0:∞)-(0:∞).^2), 1:2:∞)))
 end
-
 @simplify function *(Δ::Laplacian, P::RealSphericalHarmonic)
     P * Diagonal(mortar(Fill.((-(0:∞)-(0:∞).^2), 1:2:∞)))
+end
+
+# this does Δ^k and should probably work automatically but currently doesn't.
+function *(D::ApplyQuasiArray{SphericalCoordinate{T},2,typeof(^),Tuple{Laplacian{SphericalCoordinate{T},DomainSets.FixedUnitSphere{StaticArrays.SArray{Tuple{3},T,1,3}}},A}}, P::SphericalHarmonic) where {T,A<:Integer}
```

Comment:
@dlfivefifty  I feel like this should be something we get automatically instead of having to explicitly define it?

Reply:
This is where debugging comes in handy: we can see how derivative works.
```julia
julia> @ent D^3 * T
In *(App, b) at /Users/solver/.julia/packages/QuasiArrays/nkKFH/src/lazyquasiarrays.jl:226
 226  function *(App::ApplyQuasiMatrix{<:Any,typeof(^),<:Tuple{<:AbstractQuasiMatrix{T},<:Integer}}, b::AbstractQuasiMatrix) where T
>227      A,p = arguments(App)
 228      p < 0 && return ApplyQuasiMatrix(^,inv(A),-p)*b
 229      p == 0 && return copy(b)
 230      return A*(ApplyQuasiMatrix(^,A,p-1)*b)
 231  end
```
Unfortunately I don't see why this isn't being called...

Reply:
Ah I see, I think I know how to fix this.

Reply:
Alright, it works automatically as intended now.